src/components/Address/index.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ENS from 'ethereum-ens'
+
+function useDebounce(value, timeout) {
+  const [debounceValue, setDebounceValue] = useState(value)
+
+  useEffect(
+    function() {
+      const timer = setTimeout(function() {
+        setDebounceValue(value)
+      }, timeout)
+
+      return function() {
+        clearTimeout(timer)
+      }
+    },
+    [value]
+  )
+  return debounceValue
+}
+
+async function resolveENSName(value, provider) {
+  const ethereumRegEx = /^0x[a-fA-F0-9]{40}$/
+  const ens = new ENS(provider)
+  if (ethereumRegEx.test(value)) {
+    return await ens
+      .reverse(value)
+      .name()
+      .then(async result => {
+        const address = await ens.resolver(result).addr()
+        if (value !== address) {
+          return {
+            type: 'error',
+            data: ''
+          }
+        } else {
+          return {
+            type: 'success',
+            data: {
+              resolved: result,
+              address: address
+            }
```

Comment:
You should only need the ENS name for this case

Reply:
We are trying to avoid having different Blockies for address and ens that are same. That's why need that. Always pass the address to the Blockies while displaying the resolved value. Plus it would be nice for the parent component to have the both